server/api/posts/repository.go

```diff
@@ -1,6 +1,8 @@
 package posts
 
 import (
+	"database/sql"
```

Comment:
Why are we using `database/sql`?

I'm pretty sure we don't have this package in the repo no?

Reply:
var catID sql.NullInt64
var updateTime sql.NullTime
For creating Post  , the UpdatedAt and CategoryID as well not have inserted in query so to handle null value, I must import data/sql to use sql.NullInt64 and sql.NullTime to allow pass null value. Actually, I fix the struct of Post in CategoryID as *uint but when use row.Scan, GORM not allow to pass the null value to this pointer. It seem like that the pointer is automatically dereference so never it can assign null value to uint and always has the error can not pass NULL to uint even in the struct the type is pointer.